### DISCUSSION TITLE
GS fails to decrypt mavlink and tunnel data does not get through

### DISCUSSION THREAD

**Discussion by parijip** [svpcom/wfb-ng#307] [Questions]
I am unable to get tunnel data going, however, with an Orange-cube connected to the drone - the gs gets the packets but is unable to decrypt. I have generated the keys on the GS - and copied the same to /etc/ on both the drone and gs. 
wfb-cli gs and wfb-cli drone outputs are attached..
![wfb-cli-gs](https://github.com/svpcom/wfb-ng/assets/119396732/15713458-d941-4a78-a2a7-6bfb96bfcc92)
![wfb-cli-drone](https://github.com/svpcom/wfb-ng/assets/119396732/bd5167aa-2357-421d-840d-4e3f33cbe33e)

**Comment by parijip** [comment]
Attaching the nmcli and iw outputs from drone:
![Screenshot from 2023-09-27 21-56-06](https://github.com/svpcom/wfb-ng/assets/119396732/74849549-70e5-41cf-92a9-4955bdea5005)
wifibroadcast@drone status:
![Screenshot from 2023-09-27 22-03-45](https://github.com/svpcom/wfb-ng/assets/119396732/eae4e9b9-4283-41d0-872f-465bcc220ced)


wifibroadcast@gs status:
![Screenshot from 2023-09-27 21-57-43](https://github.com/svpcom/wfb-ng/assets/119396732/95d5f1d6-06e1-4900-9125-922743dc6373)
nmcli and iw output from gs:
![Screenshot from 2023-09-27 21-58-55](https://github.com/svpcom/wfb-ng/assets/119396732/d8c4e622-3d6b-4a9d-bb43-fa999d02bfd6)

**Comment by parijip** [comment]
The mtu of drone-wfb interface was autoset to 1443 (and not the default 1445). Used ifconfig command to set the mtu of the drone-wfb interface to 1445 to match with the gs-wfb interface on the gs, before posting the screenshots above.

**Reply by svpcom** [reply]
1443 is right mtu (radio_mtu - 2). Becase every packet prefixed with two byte length

**Reply by svpcom** [reply]
You need to use the same keypair from the both sides. I.e. if generate keypair then distribue gs.key to GS and drone.key to drone

**Reply by parijip** [reply]
I have generated the keys on gs using wfb_keygen. I copied the gs.key to /etc/ on gs and the drone.key onto /etc/ on drone.

**Reply by parijip** [reply]
Then is the mtu size showing up as 1445 on the gs side wrong? Should I force change that to 1443?

**Reply by parijip** [reply]
1 more info: the wfb-ng version on the gs is 22.11.30.42394 (as it works with another drone which is running this version of wfb-ng).
The wfb-ng version on the new drone that I am trying to setup is: 23.8.14.60981
Could this mismatch be the cause of the communication failure?

**Comment by parijip** [comment]
It is working once I changed the version of wfb-ng on the gs to 23.9.26.36320-1
